server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/BayesianAnalysis.java

```diff
@@ -294,7 +294,7 @@ public void service(Mail mail) throws MessagingException {
 
             if (ignoreLocalSender) {
                 // ignore the message if the sender is local
-                if (mail.getSender() != null && getMailetContext().isLocalServer(mail.getSender().getDomain())) {
+                if (mail.hasSender() && getMailetContext().isLocalServer(mail.getSenderAsOptional().get().getDomain())) {
```

Comment:
could be 
```java
if (mail.getSenderAsOptional()
    .map(MailAddress::getDomain)
    .map(domain -> getMailetContext().isLocalServer(domain))
    .orElse(false))
```

Reply:
Could be ^^

But we do not really care about `Bayesian*` code quality.

If everyone is Ok with it I would rather not invest too much time in that part of the code.

Reply:
Same comment as beginnig, if you have a fixup to do, include it please.

Reply:
Applied